tests/static_analyzer/program_info/test_models.py

```
@@ -0,0 +1,164 @@
+def fact(name: str, file_path: str = "pkg/module.py", detail: str = "") -> SymbolFact:
+    return SymbolFact(name, 12, file_path, 1, 4, detail=detail)
+
+
+def information(*edges: EdgeEvidence) -> ProgramInformation:
+    return ProgramInformation((fact("a"), fact("b"), fact("c", "other/c.py")), tuple(sorted(edges)))
+
+
+def test_domain_values_are_immutable():
+    symbol = fact("a")
+    with pytest.raises(FrozenInstanceError):
+        setattr(symbol, "detail", "changed")
```

Ruff (0.16.0)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)